services/twilioService.js

```diff
@@ -0,0 +1,225 @@
+
+class TwilioService {
+  constructor(hmToken, contacts, messagesCallback) {
+    this.hmToken = hmToken;
+    this.id = `${JSON.parse(hmToken)._id}`;
+
+    this.chatClient = null;
+    this.messageService = new MessageService(messagesCallback);
+
+    // Map user_id to chat channel object
+    this.channels = {};
+    this.contacts = [contacts];
+
+    this.startup(contacts);
+  }
+
+  async startup(contacts) {
+    const success = await this.loadTwilioClient();
+    if (success) {
+      // Initialize all channels
+      const doneInitChannels = await this.initAllChannels(contacts);
+      if (doneInitChannels) {
+        await this.getAllMessages();
+      } else {
+        // TODO: something bad happened
+      }
+    } else {
+      // TODO: Add failure logging
+    }
+  }
+
+  async getAllMessages() {
+    await this.contacts.map(contact => this.updateMessages(contact));
+  }
+
+  async sendMessage(contact, message) {
+    try {
+      const c = this.channels[contact];
+      c.sendMessage(message);
+    } catch (e) {
+      // TODO: add sentry logging
+    }
+  }
+
+  async initAllChannels(contacts) {
+    const success = await Promise.all(contacts.map(contact => this.initSingleChannel(contact)));
+    if (success) {
+      return true;
+    }
+    return false;
+  }
+
+  async initSingleChannel(contact) {
+    const channel = await this.getChannelForChat(contact);
+    if (channel) {
+      channel.on('messageAdded', message => this.messageService.updateLocalMessageStateSingle(contact, message));
+      this.channels[contact] = channel;
+
+      // If we are the user who created the channel, always attempt to invite the other user
+      //  We only want to do this if we created the channel, because any other user on the channel
+      //   will not have permission to invite new users
+      if (channel.createdBy === this.id) {
+        this.tryInviteContact(contact);
+      }
+
+      return true;
+    }
+    return false;
+  }
+
+  async getChannelForChat(contact) {
+    const channelName = this.getChannelName(contact);
+    return this.chatClient.getUserChannelDescriptors().then((paginator) => {
+      // If this user has channels already, check if there is a channel
+      // between current user and the user being messaged
+      const channel = paginator.items.find(currentChannel => currentChannel.uniqueName === channelName);
+      if (channel) {
+        return channel.getChannel();
+      }
+      return this.createChannelWithUser();
+    });
+  }
+
+  async loadTwilioClient() {
+    let localToken = null; // await AsyncStorage.getItem('twilio_token');
+
+    if (localToken) {
+      const clientReady = await this.initChatClient(localToken);
+      if (clientReady) {
+        return true;
+      }
+    }
+
+    localToken = await this.getTwilioToken();
+    if (localToken) {
+      await AsyncStorage.setItem('twilio_token', localToken);
+      const clientInit = await this.initChatClient(localToken);
+      return clientInit;
```

Comment:
`return await this.initChatClient(localToken);`

Reply:
Thanks, fixed in next push.